research-software-directory/RSD-as-a-service
# 106 software information right

# Edit software page - Software information

Closes #106

Changes proposed in this pull request:
* User can edit and save all information under Software Information section (see image)
* User will be notified in case it makes changes and want to navigate without saving the information
* On software page when user is maintainer "Edit" button will be shown at the top right section, next to Add and User menu.
* Cancel button is removed
* Header section is moved at the top of the page and shred between steps

How to test:
* `docker-compose down --volumes`: we clean old data, just to be sure that everything works
* `docker-compose build`: build all modules
* `docker-compose up`: start services
*  run data migration to have more data (not required)
*  open app and confirm [software is visible on the page](http://localhost/software)
*  Sign in using SURFconext test IdP, use test credentials `professor1`
*  create new software using + button at the top right corner and "New software" menu option
![image](https://user-images.githubusercontent.com/9204081/153047450-dd0ff889-ca6b-4458-8d7b-a08c83fb91e9.png)
*  define remote markdown file in edit section
*  add few `keywords`: you can select or find options by typing (no custom options are allowed)
*  `save` using save button. After saving the button will be disabled
*  add few licenses: Apache 2 and MIT for example, same approach as keywords you can select from dropdown or filter by typing. The list of licenses is [pulled from spdx repo](https://github.com/spdx/license-list-data)
* `save` button will be enabled again
*  try to navigate to diferent section, you will be notified that changes are not saved - `select cancel and stay on page`
* `save` using save button
* try again to navigate to another section, the unsaved warning `should not be show - as you saved changes`
* navigate back to software information section, all information should be present 
* turn ON publish and featured swithes, `save` button should be active again
* if you loaded large readme file you will be able to scroll down the page. The header with the software title and save button `should remain at the top of the page` like in the image below
![image](https://user-images.githubusercontent.com/9204081/153049070-27f26a5e-5c82-4818-a28a-dacbba17e8a4.png)
* try to reload the page using reload button of your browser, you should receive warning about unsaved content `select Cancel to remain on page`
![image](https://user-images.githubusercontent.com/9204081/153049576-75816ad2-dec7-4b3e-9c42-aa34ec1763b3.png)
* `save` the changes
* use `reload button` again, this time you should NOT receive any warnings (as you saved the changes at previous step)
* Go to [software overview page](http://localhost/software), if you `activated publish and featured switches` your software should be on the first page, probably at first place as it is the most recently updated software item.
![image](https://user-images.githubusercontent.com/9204081/153051125-38f41272-0425-4a34-9577-c8bcbe63463c.png)
* Select your new software, the software page should be loaded and have all info you defined (check content of remote markdown). Note! remote markdown might have broken links because it `runs in different context` from where it is pulled (github) and if relative paths are used in the links (which normally it is the case).
* Verify that you see `edit button`, at the top right corner, as in the picture below
![image](https://user-images.githubusercontent.com/9204081/153050790-0269f990-91f2-45d7-b3b6-9fdcdaf5ed0e.png)

PR Checklist:

*   [n/a] Update documentation
